cql3/expr/expression.cc

```diff
@@ -546,6 +549,9 @@ bool is_satisfied_by(const binary_operator& opr, const evaluation_inputs& inputs
                     throw exceptions::unsupported_operation_exception("LIKE lhs is subscripted");
                 } else if (opr.op == oper_t::IN) {
                     return is_one_of(sub, opr.rhs, inputs);
+                } else if (opr.op == oper_t::IS_NOT && evaluate(opr.rhs, inputs).is_null()) {
+                    cql3::raw_value val = evaluate(opr.lhs, inputs);
+                    return !val.is_null();
```

Comment:
Please add unit tests.

Reply:
I will move all `evaluate` and `is_satisfied_by` changes to a separate PR to make this one smaller.
These changes are completely independent and with unit tests it's gonna be ~10 commits already.